tycho-gpg-plugin/src/main/java/org/eclipse/tycho/gpg/SignRepositoryArtifactsMojo.java

```diff
@@ -30,213 +26,269 @@
+     * existing signature(s) with a new signature, and {@code merge} to add a new signature to any
+     * existing signature(s).
+     */
+    @Parameter(defaultValue = "skip")
+    private PGPKeyBehavior pgpKeyBehavior;
+
+    /**
+     * Configure the signer used for PGP signing. Currently supported are {@code gpg} for launching
+     * the native {@code gpg} executable, and {@code bc} for using Bouncy Castle libraries. The
+     * latter is much faster and it can sign in parallel, so is very much faster.
+     */
+    @Parameter(defaultValue = "pgp")
+    private String signer;
+
+    /**
+     * Configured to specify artifacts that should be signed independently of other settings, e.g.,
+     * {@link #skipIfJarsigned}, {@link #skipIfJarsignedAndAnchored}, and {@link #skipBinaries}.
      */
     @Parameter
     private List<String> forceSignature;
 
-    @Component(role = UnArchiver.class, hint = "xz")
-    private XZUnArchiver xzUnarchiver;
-
     @Component(role = UnArchiver.class, hint = "zip")
     private ZipUnArchiver zipUnArchiver;
 
     @Component(role = Archiver.class, hint = "xz")
     private XZArchiver xzArchiver;
 
-    @Component(role = Archiver.class, hint = "zip")
-    private ZipArchiver zipArchiver;
+    @Component(role = IProvisioningAgent.class)
+    private IProvisioningAgent agent;
+
+    @Component(role = EquinoxServiceFactory.class, hint = "tycho-core")
+    private EquinoxServiceFactory serviceFactory;
+
+    @Override
+    protected String getSigner() {
+        return signer;
+    }
+
+    @Override
+    protected File getPGPInfo() {
+        var pgpInfo = System.getProperty("org.eclipse.tycho.test.pgp.info");
+        return pgpInfo == null ? null : new File(pgpInfo);
+    }
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-        var artifactsXml = new File(repository, "artifacts.xml");
-        var artifactsXmlXz = new File(repository, "artifacts.xml.xz");
-        var artifactsJar = new File(repository, "artifacts.jar");
-        if (!artifactsXml.exists()) {
-            if (artifactsXmlXz.exists()) {
-                xzUnarchiver.setSourceFile(artifactsXmlXz);
-                xzUnarchiver.setDestFile(artifactsXml);
-                xzUnarchiver.extract();
-                artifactsXmlXz.delete();
-            }
-            if (artifactsJar.exists()) {
-                zipUnArchiver.setSourceFile(artifactsJar);
-                zipUnArchiver.setDestDirectory(repository);
-                zipUnArchiver.extract();
-                artifactsJar.delete();
+        var signedContentFactory = serviceFactory.getService(SignedContentFactory.class);
+
+        var signer = newSigner(project);
+        var keys = KeyStore.create();
+
+        var artifactRepositoryManager = agent.getService(IArtifactRepositoryManager.class);
+        try {
+            var artifactRepository = (IFileArtifactRepository) artifactRepositoryManager
```

Comment:
If it helps her you probably can use `SimpleArtifactRepositoryIO` directly instead of using the manager to load/save repository.

Reply:
The useful methods for getting the artifact location using whatever mapping rules are defined is important so no, that doesn't help.